azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ChildListFlattener.java

```diff
@@ -0,0 +1,181 @@
+package com.microsoft.azure.management.compute.implementation;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.Page;
+import com.microsoft.azure.PagedList;
+import com.microsoft.rest.RestException;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * {@link ChildListFlattener} that can take a paged list of parents and flatten their child lists
+ * as a single lazy paged list.
+ *
+ * @param <T> the type of parent paged list item
+ * @param <U> the type of child paged list item
+ */
+public final class ChildListFlattener<T, U> {
```

Comment:
Yes, it should be, will make it internal. At some point of time we might want to move this to runtime or resources, incase we encounter similar usecases in other modules